crates/codegen/runtime/cargo/crate/src/ir/builder.rs.jinja2

```diff
@@ -19,41 +19,39 @@
 
   {% for parent_type, sequence in builder.sequences %}
     {% if not sequence.has_added_fields %}
-      pub fn build_{{ parent_type | snake_case }}(node: &Rc<NonterminalNode>) -> Result<{{ parent_type }}>
+      pub fn build_{{ parent_type | snake_case }}(node: &Rc<NonterminalNode>) -> Option<{{ parent_type }}>
       {
-        expect_nonterminal_kind(node, NonterminalKind::{{ parent_type }})?;
+        assert_nonterminal_kind(node, NonterminalKind::{{ parent_type }});
         let mut helper = ChildrenHelper::new(&node.children);
         {% for field in sequence.fields %}
           {%- if field.is_removed -%}
             _ = helper.accept_label(EdgeLabel::{{ field.label | pascal_case }})?;
           {%- elif field.is_optional -%}
             let {{ field.label }} = if helper.at_label(EdgeLabel::{{ field.label | pascal_case }}) {
```

Comment:
not sure I understand why we need `at_label()` here, in addition to `accept_label()`. The latter should return `None` if the label is not found, right?
As a bonus, it will also skip any leading trivia, or error nodes.

Reply:
Same for other calls below.

Reply:
`accept_label()` will return `None` only if it found an invalid node (after skipping it).  If there were no parse errors, it will panic with the assertion on the expected label.

Reply:
As discussed offline, I removed `at_label()` in favour of skipping over invalid nodes same as trivia and returning `None` if the label doesn't match in `accept_label()`.